src/common/time/src/timezone.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::str::FromStr;
+
+use chrono::{FixedOffset, Local};
+use chrono_tz::Tz;
+use snafu::{OptionExt, ResultExt};
+
+use crate::error::{
+    InvalidTimeZoneOffsetSnafu, ParseOffsetStrSnafu, ParseTimeZoneNameSnafu, Result,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum TimeZone {
+    Offset(FixedOffset),
+    Named(Tz),
+}
+
+impl TimeZone {
+    /// Compute timezone from given offset hours and minutes
+    /// Return `None` if given offset exceeds scope
+    pub fn hours_mins_opt(offset_hours: i32, offset_mins: u32) -> Result<Self> {
+        let offset_secs = if offset_hours > 0 {
+            offset_hours * 3600 + offset_mins as i32 * 60
+        } else {
+            offset_hours * 3600 - offset_mins as i32 * 60
+        };
+
+        FixedOffset::east_opt(offset_secs)
+            .map(Self::Offset)
+            .context(InvalidTimeZoneOffsetSnafu {
+                hours: offset_hours,
+                minutes: offset_mins,
+            })
+    }
+
+    /// Parse timezone offset string and return None if given offset exceeds
+    /// scope.
+    ///
+    /// String examples are available as described in
+    /// https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html
+    ///
+    /// - `SYSTEM`
+    /// - Offset to UTC: `+08:00` , `-11:30`
+    /// - Named zones: `Asia/Shanghai`, `Europe/Berlin`
+    pub fn from_tz_string(tz_string: &str) -> Result<Option<Self>> {
+        // Use system timezone
+        if tz_string.eq_ignore_ascii_case("SYSTEM") {
+            Ok(None)
+        } else if let Some((hrs, mins)) = tz_string.split_once(':') {
+            let hrs = hrs
+                .parse::<i32>()
+                .context(ParseOffsetStrSnafu { raw: tz_string })?;
+            let mins = mins
+                .parse::<u32>()
+                .context(ParseOffsetStrSnafu { raw: tz_string })?;
+            Self::hours_mins_opt(hrs, mins).map(Some)
+        } else if let Ok(tz) = Tz::from_str(tz_string) {
+            Ok(Some(Self::Named(tz)))
+        } else {
+            ParseTimeZoneNameSnafu { raw: tz_string }.fail()
+        }
+    }
+}
+
+impl ToString for TimeZone {
```

Comment:
Impl `fmt::Display` instead of `ToString`: https://doc.rust-lang.org/std/string/trait.ToString.html

Reply:
TIL